test/test_support.ts

```diff
@@ -120,6 +120,12 @@ export function createSourceCachingHost(
     if (sources.has(fileName)) {
       return true;
     }
+    // Typescript occasionally needs to look on disk for files we don't pass into
+    // the program as a source (eg to resolve a module that's in node_modules),
+    // but only .ts files explicitly passed in should be findable
+    if (/.*\.ts$/.test(fileName)) {
+      return false;
+    }
```

Comment:
Why is this necessary in this change?

Reply:
Test support provides a kind of hermetic way to run a typescript compilation, where you explicitly pass in the sources as strings.  However, we have one test that needs to check if a file exists on disk to do module resolution inside node_modules (even though it doesn't actually read the contents of the file).

I added a test case where I wanted to compile 2 .d.ts's to check them for consistency, but without this line, tsc would notice that there was a .ts file around and try to grab that instead of the .d.ts file I wanted to compile.

This was the easiest way I could find to resolve the 2 use cases.